dosing_pump_controller/SetDoseMenuItem.cpp

```diff
@@ -0,0 +1,83 @@
+//  Created by Andrew Florjancic on 9/23/24.
+//  Copyright © 2024 Andrew Florjancic. All rights reserved.
+#include "SetDoseMenuItem.h"
+
+SetDoseMenuItem::SetDoseMenuItem(PumpController& pump_controller) : MenuItem("Set Dose", false), pump_controller(pump_controller) {
+    // Add dosing frequency options
+    frequency_menu.append(2);   // Dose every 12 hours
+    frequency_menu.append(3);   // Dose every 8 hours
+    frequency_menu.append(4);   // Dose every 6 hours
+    frequency_menu.append(6);   // Dose every 4 hours
+    frequency_menu.append(8);   // Dose every 3 hours
+    frequency_menu.append(12);  // Dose every 2 hours
+    frequency_menu.append(24);  // Dose every hour
+    frequency_menu.append(48);  // Dose every 30 minutes
+    frequency_menu.append(96);  // Dose every 15 minutes
+    frequency_menu.append(192); // Dose every 7.5 minutes
+    frequency_menu.setCurrentItem(pump_controller.getDoseFrequency());
+}
+
+void SetDoseMenuItem::showFeature() {
+    display->lcd.clear();
+    switch(feature_state) {
+        case frequency: display->lcd.print(frequency_message + String(frequency_menu.getCurrentItem()->data)); break;
+        case dose_hundreds:
+        case dose_tens:
+        case dose_ones: display->lcd.print(dose_message + String(dose_duration)); break; 
```

Comment:
Not now but later we should add some indicator to the display to highlight which position is being updated since this is currently unclear. Perhaps a simple underlined character would be fine.